src/Bicep.Core/Parser/Parser.cs

```diff
@@ -390,7 +390,7 @@ private NumericLiteralSyntax NumericLiteral()
         private SyntaxBase InterpolableString()
         {
             var startToken = reader.Peek();
-            var tokensOrSyntax = new List<TokenOrSyntax>();
+            var tokensOrSyntax = new List<SyntaxBase>();
```

Comment:
I did not look to simplify anything here btw. There are probably some opportunities to simplify, but I'm keeping that out of this PR.